### ISSUE TITLE
Error: EISDIR: illegal operation on a directory

### ISSUE THREAD

**Issue by mickeyvip** [jonycheung/deadsimple-less-watch-compiler#47]
I am experiencing the same problem with every version > `1.4` (`1.4.2` works ok).

Just set the `watchFolder` to be relative: 

`./less` 
`../src/less` (in my case)

and the exception is thrown.

Debugging the code of the latest version (`1.10`) I encountered the following code in `lessWatchCompilerUtils.js:184`:

```javascript
fileWatcher: function fileWatcher(f, files, options, filelist, fileimportlist, watchCallback) {
  if (filelist.indexOf(f) !== -1) return;
  filelist[filelist.length] = f;

  if (f.indexOf('.') !== -1) {
    fileimportlist[f] = fileSearch.findLessImportsInFile(f);
  }
  lessWatchCompilerUtilsModule.setupWatcher(f, files, options, watchCallback);
  for (var i in fileimportlist[f]) {
    if (filelist.indexOf(fileimportlist[f][i]) === -1) {
      lessWatchCompilerUtilsModule.setupWatcher(path.normalize(path.dirname(f) + '/' + fileimportlist[f][i]), files, options, watchCallback);
    }
  }
}
```

this function is called (probably the first time) with `f` set to the `watchFolder` and the second `if` (`if (f.indexOf('.') !== -1)`) will resolve to `true` if `f` has `.` inside it (and every relative folder name does) calling `fileSearch.findLessImportsInFile(f)` which will fail by calling `fs.readFileSync(f, 'utf8')` in `filesearch.js:15`.

The condition `if (f.indexOf('.') !== -1)` if pretty fragile one. Why not checking if `f` is a directory?

Changing the condition to `if (f.lastIndexOf('.') > 2)` solves the problem for me, but it's a hack.

Thank you.

**Comment by jonycheung** [comment]
Thanks for the great details. I'll look into this asap.

**Comment by jonycheung** [comment]
@mickeyvip It looks like you may have debugged it a fair bit. Would you be interested in submitting a PR?

**Comment by mickeyvip** [comment]
@jonycheung thanks, I would love to submit a PR.

I just want to be sure what exactly `if (f.indexOf('.') !== -1)` is supposed to check.

**Comment by jonycheung** [comment]
yes, that's supposed to be checking for files (not directories) before trying to process it. That's definitely fragile.

**Comment by mickeyvip** [comment]
@jonycheung, the PR is ready

**Comment by mickeyvip** [comment]
@jonycheung , should this be closed?

**Comment by jonycheung** [comment]
Just released `v1.10.2` with the fix. Let me know if that doesn't work.

**Comment by mickeyvip** [comment]
it doesn't fail now.
Thanks!
